### DISCUSSION TITLE
Masowa zmiana czasu dostawy w ofertach

### DISCUSSION THREAD

**Discussion by lukiopmoto** [allegro/allegro-api#8381] [Suggestions - resolved]
Witam, 

Mamy 10 tysięcy aukcji. Realizujemy wysyłki 3x w tygodniu (w poniedziałek, wtorek i środę). W tych dniach czas dostawy ustawiamy na PT24H. W czwartek i piątek wysyłki nie są realizowane, więc nagle w środę od godziny 12:00 musimy zmieniać czas dostawy na PT72H. Obecnie zmiana taka wymaga 10 tysięcy operacji, tj. pojedynczych zmian dla każdej oferty co jest po prostu mega niewydajne. 

Chciałbym zrobić to tak jak wykonuje się inne operacje zbiorcze podając 100 czy 500 identyfikatorów ofert oraz nowy czas dostawy i zmienia się to jednym requestem tak jak to:

 $data = array(
            "modification" => array (
                "delivery" => array(
                    "HandlingTime" => "PT48H"
                )
            ),
            "offerCriteria" => array(
                array(
                    "offers" => array (
                        array("id" => '12345'), array("id" => '34567'), array("id" => '98765')
                    ),
                    "type" => "CONTAINS_OFFERS"
                )
            )
        );

Zmiana czasu dostawy obecnie (pojedynczo) wykonuje się obecnie kilka godzin, przez co nie wszystkie oferty zdążą się zaktualizować i część klientów kupuje towar z myślą iż dostanie go następnego dnia.

**Comment by omegapraca** [comment]
U nas ta sama sytuacja. Pojedyncza edycja setek a czasami tysięcy ofert która zajmuje czasami kilka godzin, przez API. Edycja czasu dostawy dla jednej oferty polega obecnie na poraniu całej oferty, zmianie czasu dostawy, walidacji na nowo całej oferty przez Allegro i zapis. Cała operacja trwa ok 1 sekundy i tak 5000x. Łatwo policzyć. Bezsensowne obciążanie nie tylko swoich zasobów, ale też Allegrowych.

**Comment by tomalm37** [comment]
We can't wait for it to finally be ready. We need it to publish more offers! How is it possible that this isn't there yet?

**Comment by GreGo69** [comment]
Również mam podobny problem ze zmianą czasu wysyłki dla kilkuset ofert.

**Comment by MaciejFrackowiak** [comment]
Wdrożyliśmy dzisiaj [możliwość grupowej zmiany czasu wysyłki w ofertach](https://github.com/allegro/allegro-api/issues/9112).
